### PR TITLE
Fix rescheduling updating calendars & duplicate bookings

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Reschedule a booking with a destination calendar

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/core/EventManager.ts [hunk 1]
```diff
@@ -439,7 +461,7 @@ export default class EventManager {
         });
       }
 
-      if (newBooking) {
+      if (newBooking?.references.length) {
```
Comment (joeauyeung) [inline packages/core/EventManager.ts:464]: Only look through the new booking, if references are created

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -1650,7 +1650,7 @@ async function handler(
 
     // Use EventManager to conditionally use all needed integrations.
 
-    const updateManager = await eventManager.reschedule(evt, originalRescheduledBooking.uid, booking?.id);
```
Comment (joeauyeung) [inline packages/features/bookings/lib/handleNewBooking.ts:1653]: The third parameter `newBookingId` is only used when rescheduling seated bookings. This was causing seated logic on normal bookings

File: packages/core/EventManager.ts [hunk 3]
```diff
@@ -279,6 +279,27 @@ export default class EventManager {
...
+        bookingId: booking.id,
+      },
+    });
+    const attendeeDeletes = prisma.attendee.deleteMany({
+      where: {
+        bookingId: booking.id,
+      },
+    });
+
+    const bookingDeletes = prisma.booking.delete({
```
Comment (emrysal) [inline packages/core/EventManager.ts:294]: Only this delete is needed, attendee & bookingReference are in Cascade and will be auto-deleted 👍

**Review by emrysal — DISMISSED** [review]
Do we want to delete the original booking on reschedule? Why doesn't a reschedule just update the same booking at a different time 😕 .

File: packages/core/EventManager.ts (hunk 3 above)
Comment (joeauyeung) [reply packages/core/EventManager.ts]: I realize that isn't the right thing to do as well. Just pushed the change

File: packages/core/EventManager.ts [hunk 4]
```diff
@@ -423,6 +423,7 @@ export default class EventManager {
     booking: PartialBooking,
     newBookingId?: number
   ): Promise<Array<EventResult<NewCalendarEventType>>> {
+    newBookingId = undefined;
```
Comment (emrysal) [inline packages/core/EventManager.ts:426]: I don't know; are you still working on this? Setting this to undefined makes all kinds of dead code - please draft if so 👍

Comment (joeauyeung) [reply packages/core/EventManager.ts:426]: Oh that's just me experimenting. Thanks for catching

File: packages/core/EventManager.ts [hunk 5]
```diff
@@ -439,7 +439,7 @@ export default class EventManager {
         });
       }
 
-      if (newBooking) {
+      if (newBooking?.references.length) {
         calendarReference = newBooking.references.find((reference) => reference.type.includes("_calendar"));
       } else {
         // Bookings should only have one calendar reference
```
Comment (alishaz-polymath) [inline packages/core/EventManager.ts:445]:
```suggestion
       calendarReference = newBooking?.references.length
         ? newBooking.references.find((reference) => reference.type.includes("_calendar"))
         : booking.references.find((reference) => reference.type.includes("_calendar"));
```
Minor NIT suggestion

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 6]
```diff
@@ -1650,7 +1650,7 @@ async function handler(
...
+    const updateManager = await eventManager.reschedule(evt, originalRescheduledBooking.uid);
```
Comment (alishaz-polymath) [inline packages/features/bookings/lib/handleNewBooking.ts:1653]: How does this change handle rescheduling for seated bookings? Or is that being handled elsewhere?

**Review by alishaz-polymath — APPROVED** [review]
Apart from NITs, it looks good. Tested locally, works

File: packages/features/bookings/lib/handleNewBooking.ts (hunk 6 above)
Comment (joeauyeung) [reply packages/features/bookings/lib/handleNewBooking.ts:1653]: It's being handled in the reschedule method, having that third parameter triggers that logic to run.

**Review by emrysal — APPROVED** [review]
LGTM :shipit:

**Pull request merged**